events/migrations.txt

```diff
@@ -4,32 +4,59 @@ namespace = migrations
+
 	trigger = {
-		base_manpower = 2 									# does not let the last rural pop to leave, obviously
 		has_global_flag = popgrowth_running					# to only affect if the popgrowth mechanic is active
-		NOT = { check_variable = { PG_trb_grow = 100 } }	# negative growth, anything under 1.00 (100%)
-		NOT = { has_province_modifier = rural_pause }		# condition that initial values are kept for 100 years
-
 	}
 
+	option = {
+		name = "migrations.1.1"
 
-	mean_time_to_happen = {
-		months = 240 	# twenty years to move for a stabilized population, will be affected by modifiers
+		every_owned_province = {
+			limit = {
+				base_manpower = 2 									# does not let the last rural pop to leave, obviously
+				NOT = { check_variable = { PG_trb_grow = 100 } }	# negative growth, anything under 1.00 (100%)
+				NOT = { has_province_modifier = rural_pause }		# condition that initial values are kept for 100 years
+			}
+			random_list = {
+				19 = {
+					# Reasons to leave. Devastation will NOT appear here, since it already influences growth.
+					modifier = { 	factor = 5.0 	check_variable = { PG_com_grow = 110 } } 			# not if there's work in the city, same province
+					modifier = {	factor = 0.65	check_variable = { PG_trb_grow = 60 }	NOT = { check_variable = { PG_trb_grow = 80 } }	}
+					modifier = {	factor = 0.40	check_variable = { PG_trb_grow = 50 }	NOT = { check_variable = { PG_trb_grow = 60 } }	}
+					modifier = {	factor = 0.25	check_variable = { PG_trb_grow = 40 }	NOT = { check_variable = { PG_trb_grow = 50 } }	}
+					modifier = { 	factor = 0.05 	NOT = { check_variable = { PG_trb_grow = 40 } } } 	# they must be really miserable...
+				}
+				1 = {
+					province_event = { id = migrations.1 }
+				}
+			}
+		}
+	}
+}
+
+province_event = {
+	id = migrations.1
+	title = "migrations.1.t"
+	desc = "migrations.1.d"
+	picture = peasant_woman
+	hidden = yes
 
-		# Reasons to leave. Devastation will NOT appear here, since it already influences growth.
-		modifier = { 	factor = 5.0 	check_variable = { PG_com_grow = 110 } } 			# not if there's work in the city, same province
-		modifier = {	factor = 0.65	check_variable = { PG_trb_grow = 60 }	NOT = { check_variable = { PG_trb_grow = 80 } }	}
-		modifier = {	factor = 0.40	check_variable = { PG_trb_grow = 50 }	NOT = { check_variable = { PG_trb_grow = 60 } }	}
-		modifier = {	factor = 0.25	check_variable = { PG_trb_grow = 40 }	NOT = { check_variable = { PG_trb_grow = 50 } }	}
-		modifier = { 	factor = 0.05 	NOT = { check_variable = { PG_trb_grow = 40 } } } 	# they must be really miserable...
+	is_triggered_only = yes
 
+	trigger = {
+		base_manpower = 2 									# does not let the last rural pop to leave, obviously
+		has_global_flag = popgrowth_running					# to only affect if the popgrowth mechanic is active
+		NOT = { check_variable = { PG_trb_grow = 100 } }	# negative growth, anything under 1.00 (100%)
+		NOT = { has_province_modifier = rural_pause }		# condition that initial values are kept for 100 years

```

Comment:
You can delete everything in this trigger, we already know it's all true because it was called from migrations.101